gobblin-modules/gobblin-eventhub/src/main/java/gobblin/eventhub/writer/EventhubSyncDataWriter.java

```diff
@@ -0,0 +1,205 @@
+
+
+  public final static String  EVH_NAMESPACE = "eventhub.namespace";
+  public final static String  EVH_HUBNAME = "eventhub.hubname";
+  public final static String  EVH_SAS_KEYNAME = "eventhub.sas.keyname";
+  public final static String  EVH_SAS_KEYVALUE = "eventhub.sas.keyvalue";
+
+  private final String namespaceName;
+  private final String eventHubName;
+  private final String sasKeyName;
+  private final String sasKey;
+  private final Timer timer = new Timer();
+  private final Meter bytesWritten = new Meter();
+
+  /** User needs to provide eventhub properties */
+  public EventhubSyncDataWriter (Properties properties) {
+    namespaceName = properties.getProperty(EVH_NAMESPACE);
+    eventHubName =  properties.getProperty(EVH_HUBNAME);
+    sasKeyName = properties.getProperty(EVH_SAS_KEYNAME);
+    sasKey = properties.getProperty(EVH_SAS_KEYVALUE);
+  }
+
+  /**
+   * Write a whole batch to eventhub
+   */
+  public WriteResponse write (Batch<JsonObject> batch) throws IOException {
+    long before = System.nanoTime();
+    String encoded = encodeBatch(batch);
+    bytesWritten.mark(encoded.getBytes().length);
+    post (encoded);
+    timer.update(System.nanoTime() - before, TimeUnit.NANOSECONDS);
+    return WriteResponse.EMPTY;
+  }
+
+  /**
+   * Write a single record to eventhub
+   */
+  public WriteResponse write (JsonObject record) throws IOException {
+    long before = System.nanoTime();
+    String encoded = encodeRecord(record);
+    bytesWritten.mark(encoded.getBytes().length);
+    post (encoded);
+    timer.update(System.nanoTime() - before, TimeUnit.NANOSECONDS);
+    return WriteResponse.EMPTY;
+  }
+
+  /**
+   * Total bytes that have been written
+   */
+  public long bytesWritten() {
+    return bytesWritten.getCount();
+  }
+
+  /**
+   * A signature which contains the duration.
+   * After the duration is expired, the signature becomes invalid
+   */
+  private String getSignature() {
+    // generate signature
+    try {
+      String signature = SharedAccessSignatureTokenProvider
+          .generateSharedAccessSignature(sasKeyName, sasKey, namespaceName, Duration.ofMinutes(45));
+      return signature;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Send an encoded string to the Eventhub using post method
+   */
+  private CloseableHttpResponse post (String encoded) throws  IOException {
+
+    String uri = "https://" + namespaceName + ".servicebus.windows.net/" + eventHubName + "/messages";
+    HttpPost httpPost = new HttpPost(uri);
+    httpPost.setHeader("Content-type", "application/vnd.microsoft.servicebus.json");
+    httpPost.setHeader("Authorization", getSignature());
+    httpPost.setHeader("Host", namespaceName + ".servicebus.windows.net ");
+
+
+    StringEntity entity = new StringEntity(encoded);
+    httpPost.setEntity(entity);
+
+    CloseableHttpResponse response = httpclient.execute(httpPost);
+
+    try {
```

Comment:
Do you care about the body of the response or just the status of the response (if it is successful)